app/controllers/users/nbp_wallet_controller.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'rqrcode'
+
+module Users
+  class NbpWalletController < ApplicationController
+    skip_before_action :require_user!
+    skip_after_action :verify_authorized
+    before_action :set_provider_details
+    before_action :set_connector_api
+
+    def connect
```

Comment:
Do you think it would make sense to check for a pending relationship template before rendering the page? If we do so, we could redirect to `finalize` before showing the QR code. This might happen if someone opens the `connect` page, scans the QR code, and closes the page before submitting their personal details through the Wallet app.

Reply:
That's a good idea. I added that.